realm/src/main/java/io/realm/BaseRealm.java

```diff
@@ -137,6 +137,17 @@ public boolean isAutoRefresh() {
     }
 
     /**
+     * Indicates if the Realm is currently in a write transaction.
+     * @return the write transaction status.
+     * @throws java.lang.IllegalStateException if the Realm is not opened or in a different thread
```

Comment:
I would remove this one. We don't have this exception listed on our other public methods either.